SubEthaEdit-Mac/Source/TCMBEEP/TCMBEEPProfile.m

```diff
@@ -27,48 +26,40 @@ - (void)handleInitializationData:(NSData *)aData {
     DEBUGLOG(@"BEEPLogDomain",DetailedLogLevel,@"%s %@ should Handle data:%@",__FUNCTION__,_profileURI,[[NSString alloc] initWithData:aData encoding:NSISOLatin1StringEncoding]);
 }
 
-- (TCMBEEPSession *)session 
-{
+- (TCMBEEPSession *)session  {
     return [[self channel] session];
 }
 
-- (BOOL)isServer
-{
+- (BOOL)isServer {
     return ![[self channel] isInitiator];
 }
 
-- (void)processBEEPMessage:(TCMBEEPMessage *)aMessage
-{
+- (void)processBEEPMessage:(TCMBEEPMessage *)aMessage {
     NSLog(@"You should have overridden this!");
 }
 
-- (void)close
-{
+- (void)close {
     I_isClosing = YES;
     [[self channel] close];
 }
 
-- (void)channelDidReceiveCloseRequest
-{
+- (void)channelDidReceiveCloseRequest {
     DEBUGLOG(@"BEEPLogDomain", SimpleLogLevel, @"channelDidReceiveCloseRequest: %@", NSStringFromClass([self class]));
 }
 
-- (void)channelDidClose
-{
+- (void)channelDidClose {
     DEBUGLOG(@"BEEPLogDomain", SimpleLogLevel, @"channelDidClose: %@ delegate:%@", NSStringFromClass([self class]), [self delegate]);
-    id delegate = [self delegate];
+    __strong typeof(_delegate) delegate = [self delegate];
```

Comment:
generally speaking this should be the pattern everywhere, as otherwise the analyzer should bark because of repeated access of weak variables. Seems like we dont' have that warning on yet though…